Added sounds for player death, player turning in air and wall danger state

aleks-papushin opened the pull request.

aleks-papushin committed:
Refactoring and new sounds addition:
- added sounds for player death, wall danger and turning in air
- refactored needful classes
- player death sounds were added to MusicPlayer

aleks-papushin committed:
- added sounds for wall danger state and
- changed sound for player turning in air

MaXal commented:
Do we really want `wav` files? They are huge. Maybe just compress to `ogg`?

MaXal commented on Assets/Scripts/Wall/WallSound.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public void PlayDangerSound()
+    {
+        _source.PlayOneShot(_dangerSound, _defaultVolume);
```
We can use `AudioSource.PlayClipAtPoint(_dangerSound, Camera.main.transform.position, _defaultVolume)`. Such we avoid creation of AudioSource.

MaXal commented on Assets/Scripts/Wall/WallSound.cs:
```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+public class WallSound : MonoBehaviour
```
Maybe it's easier to just implement this in `WallDanger`? It shouldn't complicated the class a lot - 1 serialized field, 1 constant and one call.

MaXal commented on Assets/Scripts/MusicPlayer.cs:
```diff
@@ -25,10 +30,13 @@ private void Awake()
-        _currentVolume = _source.volume;
-        PlayLoop(_source, _intro);
+        _musicSource = gameObject.AddComponent<AudioSource>();
+        _soundSource = gameObject.AddComponent<AudioSource>();
```
We can avoid creation of `AudioSource` by just playing with `AudioSource.PlayClipAtPoint`.
`_musicSource` on the other hand should stay to allow `PlayLoop`. What benefits bring the dynamic creation instead of the one from Unity editor?

aleks-papushin replied:
Nice, changed

aleks-papushin committed:
Changes by review comments:
- removed WallSound
- audio clips are preferrably played via AudioSource.PlayClipAtPoint()
- dynamically assigned AudioSources replaced by assigned in inspector

aleks-papushin commented:
"Do we really want wav files? They are huge. Maybe just compress to ogg?"
I agree but let's don't do this for previously imported sounds now. Just take as remark for future
All other comments are good too and fixed
